apps/genpkey.c

```diff
@@ -103,10 +103,36 @@ static void show_gen_pkeyopt(const char *algname, OSSL_LIB_CTX *libctx, const ch
     EVP_PKEY_CTX_free(ctx);
 }
 
+int write_to_file(BIO *in, const char *filename, int format, int private)
+{
+    int rv = 0, ret = 0;
+    BIO *out = NULL;
+    BUF_MEM *mem_buffer = NULL;
+
+    rv = BIO_get_mem_ptr(in, &mem_buffer);
+    if (rv <= 0) {
+        BIO_puts(bio_err, "Error reading mem buffer\n");
+        goto end;
+    }
+    out = bio_open_owner(filename, format, private);
+    if (out == NULL)
+        goto end;
+    rv = BIO_write(out, mem_buffer->data, mem_buffer->length);
+    if (rv < 0 || (size_t)rv != mem_buffer->length)
+        BIO_puts(bio_err, "Error writing to output file\n");
+    else
+        ret = 0;
```

Comment:
```suggestion
        ret = 1;
```

Reply:
So far, the function never returns success.

Reply:
Updated.